bump: Add release notes for v1.3.0, v1.3.1, add roadmap for v1.4.0 (#219)
networkservicemesh/site: content/docs/releases/v1.3.0.md

## Kubernetes Compatibility
NSM v1.3.0 has been successfully [integration tested](https://github.com/networkservicemesh/integration-k8s-kind/runs/6015599705?check_suite_focus=true) with Kubernetes versions:

- v1.23.0
- v1.22.1
- v1.21.1
- v1.20.7
- v1.19.11
- v1.18.15

## Public Cloud Capability

NSM v1.3.0 has been tested on [GKE](https://github.com/networkservicemesh/integration-k8s-gke/runs/6015598266?check_suite_focus=true), [AKS](https://github.com/networkservicemesh/integration-k8s-aks/runs/6015599486?check_suite_focus=true), [EKS](https://github.com/networkservicemesh/integration-k8s-aws/runs/6015597828?check_suite_focus=true), and on vanilla K8s on [bare metal in Equinix Metal](https://github.com/networkservicemesh/integration-k8s-packet/runs/6015598137?check_suite_focus=true),
[Cross public clusters](https://github.com/networkservicemesh/integration-interdomain-k8s/runs/6030822307?check_suite_focus=true).

## Changes since v1.2.0

### vL3

Implementation of a virtual Layer 3 (vL3) Network Service(NS) that can be used to provide a common routing domain (or vrf) to which workloads in different clusters/cloud providers may attach.

> Example: https://github.com/networkservicemesh/deployments-k8s/tree/release/v1.3.0/examples/features/vl3

Reviewer: ```suggestion
[Try NSM vL3 Example](https://github.com/networkservicemesh/deployments-k8s/tree/release/v1.3.0/examples/features/vl3)
```

Author: Thanks, applied

### NSM+Calico-VPP Integration

Allow NSM to utilize external VPP instances it did not create itself.  This would allow NSM and Calico-VPP to share a single
VPP instance on a Node, improving performance in that case.

Note: You need only to change the CNI and NSM will work without any additional changes.

> Examples for CNI Calico: https://github.com/networkservicemesh/deployments-k8s/tree/release/v1.3.0

Reviewer: ```suggestion
[Try NSM using the same VPP as Calico-CNI](https://github.com/networkservicemesh/deployments-k8s/tree/release/v1.3.0)
```

Author: Thanks, applied

### Enable Mutually Aware NSEs to overlap in IPAM/Routes

NSM by default presumes that all NSEs are 'mutually ignorant', meaning they do not know about each other.  For this reason,
if an NSM currently precludes 'overlap' in the assigned IPs and requested routes between the various NSEs to which a NSC
is connected.  Enabling Mutually Aware NSEs allows this limitation to be loosened in the case that two or more NSEs

> signal mutual awareness.  This eanbles certain 'multi-homing' cases.

Reviewer 2: ```suggestion
signal mutual awareness.  This enables certain 'multi-homing' cases.
```

> Example: https://github.com/networkservicemesh/deployments-k8s/tree/release/v1.3.0/examples/features/mutually-aware-nses

Reviewer: ```suggestion
[Try Mutually Aware NSE with overlap in IPAM/Routes Example](https://github.com/networkservicemesh/deployments-k8s/tree/release/v1.3.0/examples/features/mutually-aware-nses)
```

Author: Thanks, applied
